crates/common/mqtt_channel/src/session.rs

```diff
@@ -40,7 +42,41 @@ pub async fn init_session(config: &Config) -> Result<(), MqttError> {
             }
 
             Err(err) => {
-                eprintln!("Connection Error {}", err);
+                match err {
+                    rumqttc::ConnectionError::ConnectionRefused(ConnectReturnCode::BadClientId) => {
+                        warn!("Failed to initialize the session with MQTT Broker due to bad client id");
+                    }
```

Comment:
The `init` function is returning a `Result<(), MqttError>`, so it's weird to log the errors  and to return `()`.

These error messages are a real improvement but they must be captured by an `Error` type.